docs/release_process.md

```diff
@@ -7,9 +7,50 @@ This document describes the bpftrace release process.
 We choose to follow semantic versioning. Note that this doesn't matter much for
 major version < 1 but will matter a lot for >= 1.0.0 releases.
 
-See https://semver.org/ .
+See https://semver.org/.
 
-## Branching model
+## Release cadence
+
+bpftrace is released twice a year. Since our biggest dependency, which also
+tends to break things, is LLVM, we align with the [LLVM release
+schedule](https://llvm.org/docs/HowToReleaseLLVM.html). In particular, a
+bpftrace release should happen **two weeks after a major LLVM release**.
+
+In addition, four weeks before the bpftrace release, we create a stabilized
+release branch, which will only receive bug fixes from that moment.
+
+Overview of the release cadence is:
+
+| Task                   | Approximate date                    | Details                                                              |
+| ---------------------- | ----------------------------------- | -------------------------------------------------------------------- |
+| release branch created | **2 weeks before the LLVM release** | [Creating a release branch](#creating-a-release-branch)              |
+| LLVM release           | usually second Tuesday of Mar/Sep   | [LLVM release schedule](https://llvm.org/docs/HowToReleaseLLVM.html) |
+| bpftrace release       | **2 weeks after the LLVM release**  | [Tagging a release](#tagging-a-release)                              |
+
+## Creating a release branch
+
+A release branch should be created one month before the planned bpftrace
+release. From that moment, only relevant bug fixes should be backported to the
+branch. Similarly, after the release is done, backport PRs should be opened
+against this branch.
+
+The purpose of the branch is to give sufficient time to test features in the
+upcoming rebase without blocking the development on the master branch.
+
+When creating a branch, the following steps (submit PR where applicable) should
+be performed:
+1. Create a new branch according to the [Branching model](#branching-model).
+1. Update Nixpks to the latest version to get the latest (pre-release) LLVM by
+   running
+   ```
+   nix flake update
+   ```
+   and committing the `flake.lock` changes to the repo.
+1. Bump the supported LLVM version in `CMakeLists.txt` and `flake.nix`, resolve
+   any potential issues, and add a CI job to `.github/workflows/ci.yml` for the
+   new version.
```

Comment:
These steps happen ~2 weeks after the branch is cut, right? The instructions make it sound like it should happen when you cut the release branch.

Reply:
That's not how I read it. I read it as this happens (meaning the creation of the release branch) 4 weeks before the actual release, which is 2 weeks before the LLVM release.

Reply:
At the time of the branch cut, unstable Nix will have LLVM .rc3 which will be pretty similar to the final release so this step can happen immediately after the branch cut to start testing early.

OTOH, it makes sense to update to released LLVM once it's out so let me add a new step to do that after the LLVM release.